src/hardware/sblaster.cpp

```diff
@@ -921,10 +953,14 @@ static uint32_t read_dma_16bit(const uint32_t bytes_to_read, const uint32_t i =
 	return check_cast<uint32_t>(bytes_read);
 }
 
-static void enqueue_frames(std::unique_ptr<AudioVector> frames)
+// This keeps track of the balance of queued frames. It's a balance because
+// it's decremented through the passage of time (every per-frame tick).
+size_t queued_frames_balance = 0;
+
+void enqueue_frames(std::vector<AudioFrame>& frames)
 {
-	frames_added_this_tick += frames->num_frames;
-	soundblaster_mixer_queue.NonblockingEnqueue(std::move(frames));
+	assert(sblaster);
+	queued_frames_balance += sblaster->output_queue.NonblockingBulkEnqueue(frames);
```

Comment:
Yep, sounds like a bug.

Reply:
Didn't have much time lately; but now getting spooled back up on this. Thanks for the reviews.

In the per-frame generating cases (the DAC as well as demos and games that setup a single frame DMA transfer count), this count is unnecessary because the PIC event callback is run at exactly the SB's playback rate. 

(Will refer back to this per-frame callback in the next comment!)